### ISSUE TITLE
Track number sorting, disc number support

### ISSUE THREAD

**Issue by for-coursera** [Beinsezii/ompl#5]
If launched with `-s "<track> - <title>"`, then track sorting is not numeric:

![](https://i.imgur.com/mBNzaZO.png)

Also, if an album consists of two discs, it seems it's impossible to separate those (tried launching with `-s "<track> - ><title>" disc` but the `disc` tag doesn't look to be recognized):

![](https://i.imgur.com/wQyXyLY.png)

Otherwise really love your player, its flexibility is very promising :)

**Comment by Beinsezii** [comment]
Yea sorting is pretty bad right now. Its basically the same sorting a shell uses, so numbers are sorted by first digit and anything Unicode like accented characters will show last. I'll see if there's a library or function for "human" sorting.

As far as disc is concerned, I probably gave it the wrong human name. For now, you can use the ID3 frame name. My guess is it's either `MCDI` or `TPOS`. You can run `ompl print track` to list all the metadata of the playing track in both human and frame names, so it should be fairly apparent which is used for disc. Use id3 frames like you would a human readable tag `ompl main ~/Music -s '<mcdi|<mcdi> - > <title>'`

Basically I couldn't find a standardized list of "human names" for the frames so I just made up my own based on their descriptions from id3.org, which now seems like a dead site. I'll probably just copy ffmpeg's instead.

**Comment by for-coursera** [comment]
It was `tpos`, indeed :)

Thank you for the help, and I probably leave this open for now until further info on the sorting issue?

Thanks again!

**Comment by Beinsezii** [comment]
Fixed in 792b053

Feel free to give it a try before 0.8.0 releases, as the new backend is decently stable right now. Just don't try to play mp4 files or anything weird like that.

**Comment by for-coursera** [comment]
@Beinsezii 

Thanks! Works as advertised :)

BTW, I noticed you added a seekbar, and not only it  looks great, but you can also navigate it with a mouse (like, skipping a long silent intro, for example :). Any plans to add hotkeys to jump a song back and forth with some predefined time period, may be? Like, 5s or 10s?

**Comment by Beinsezii** [comment]
Yes hotkeys and CLI are coming. I only just added the seekbar a few commits before I fixed the tag issue.

Also I know that I said the new backend was "decently stable" but about 5 minutes before you wrote that comment I found a pretty bad bug where playing the previous track would cause the audio to play twice over itself. It's partially fixed in commit b6d96f1.

Alternatively you can build the master branch with  `--no-default-features --features=media-controls,tui,clipboard,backend-rodio` to use the old backend and still have the new tagging features. You'll lose the seekbar though.

**Comment by for-coursera** [comment]
Thanks again, you are the best! :)

**Comment by Beinsezii** [comment]
Out of curiosity about how long does it take for the seekbar to first appear for you? I'm using a rather unconventional method to implement it, and while it's performing well on my $2500 computer that's hardly a representative result.

**Comment by for-coursera** [comment]
My first track in the library is rather long, around 23 minutes, and the laptop I'm using `ompl` on is quite old, Thinkpad X260, so, yes, it takes some 5s to appear :)

**Comment by Beinsezii** [comment]
@for-coursera 
> Any plans to add hotkeys to jump a song back and forth with some predefined time period, may be? Like, 5s or 10s?

Implemented in 7f0ea62036cec3f37d139d0704db5dca4245d8e2

**Comment by for-coursera** [comment]
@Beinsezii 

Thank you!
